wear/src/main/java/io/homeassistant/companion/android/home/views/SensorUi.kt

```diff
@@ -21,13 +25,46 @@ fun SensorUi(
     basicSensor: SensorManager.BasicSensor,
     onSensorClicked: (String, Boolean) -> Unit,
 ) {
+    var checked = sensor?.enabled == true
+
+    val backgroundRequest =
+        rememberLauncherForActivityResult(ActivityResultContracts.RequestPermission()) {
+            checked = sensor?.enabled == true && it
+        }
+
+    val permissionLaunch = rememberLauncherForActivityResult(
+        ActivityResultContracts.RequestMultiplePermissions()
+    ) { isGranted ->
+        var allGranted = true
+        isGranted.forEach {
+            if (
+                manager.requiredPermissions(basicSensor.id).contains(Manifest.permission.ACCESS_FINE_LOCATION) &&
+                manager.requiredPermissions(basicSensor.id).contains(Manifest.permission.ACCESS_BACKGROUND_LOCATION) &&
+                it.key == Manifest.permission.ACCESS_FINE_LOCATION && Build.VERSION.SDK_INT >= Build.VERSION_CODES.R
+            )
+                backgroundRequest.launch(Manifest.permission.ACCESS_BACKGROUND_LOCATION)
```

Comment:
Its required to be in the loop because on certain devices anytime fine location is requested we also request background location and we need to detect if we are ready for background permissions after fine location was granted.

Reply:
All permissions are already granted when the callback is called, it isn't required to loop over them 😉

Now that the return is added it is fine, you'd still need a loop to check if everything is granted.